Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -169,5 +162,50 @@ - (nonnull NSArray *)filterResponseInterceptors:(nonnull NSArray *)proposedRespo
     return [NSArray arrayWithArray:responseInterceptors];
 }
 
+- (void)processResponse:(NSURLResponse *)response onThread:(NSThread *)thread
+{
+    __block CDTHTTPInterceptorContext *ctx =
+    [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    ctx.response = (NSHTTPURLResponse*)response;
+    for (NSObject<CDTHTTPInterceptor> *obj in self.responseInterceptors) {
+        ctx = [obj interceptResponseInContext:ctx];
+    }
+
+    if (ctx.shouldRetry && self.remainingRetries > 0) {
+        // retry
+        self.remainingRetries--;
+        self.inProgressTask = [self makeRequest];
+        [self.inProgressTask resume];
+    } else {
+        MYOnThread(thread, ^{
+            [self.delegate receivedResponse:response];
+        });
+        self.finished = YES;
+    }
+}
+
+- (void)processError:(NSError *)error onThread:(NSThread *)thread
```

Comment:
Indeed, Rhys is correct that `processError:onThread` deals with lower level networking errors, so I believe the overall handling of response interceptors in case of errors is unchanged - i.e. HTTP error codes will be handled in the `processResponse:onThread:` method and `processError:onThread:` deals with lower level networking errors, but in both cases the interceptors are called if appropriate.